DSharpPlus/Entities/DiscordMessage.cs

```diff
@@ -308,14 +314,15 @@ internal DiscordMessageReference InternalBuildMessageReference()
                     Id = guildId.Value,
                     Discord = client
                 };
-
-            var channel = client.InternalGetCachedChannel(channelId);
+            if (channelId.HasValue) return null;
```

Comment:
This check doesn't make sense, and a message reference can still be built with a message id.

Reply:
Oh wait is the channel Id nullable? I guess I didn't check, or misread what you said when I asked when about how to go about it